src/cargo/ops/cargo_clean.rs

```diff
@@ -243,6 +243,80 @@ fn clean_specs(
+                let path_dot: &str = &format!("{crate_name}.");
+                let path_dash: &str = &format!("{crate_name}-");
+
+                // Iterate over all compile modes to cover all possible hashed files.
+                for &mode in &[
+                    CompileMode::Build,
+                    CompileMode::Test,
+                    CompileMode::Check { test: false },
+                ] {
+                    for (compile_kind, layout) in &layouts {
+                        let examples_dir = layout.build_dir().examples();
+                        let examples_dir_glob = escape_glob_path(examples_dir)?;
+
+                        let triple = target_data.short_name(compile_kind);
+                        let (file_types, _) = target_data.info(*compile_kind).rustc_outputs(
+                            mode,
+                            target.kind(),
+                            triple,
+                            clean_ctx.gctx,
+                        )?;
+
+                        // Clean hashed example artifacts.
+                        for file_type in file_types {
+                            let hashed_name = file_type.output_filename(target, Some("*"));
+                            clean_ctx
+                                .rm_rf_glob(&Path::new(&examples_dir_glob).join(&hashed_name))?;
+                        }
+
+                        // Clean rustc-generated depinfo and debug files.
+                        let mut glob_pattern = examples_dir_glob;
+                        if !glob_pattern.ends_with(std::path::MAIN_SEPARATOR) {
+                            glob_pattern.push(std::path::MAIN_SEPARATOR);
+                        }
+                        glob_pattern.push('*');
+
+                        let paths = [
+                            // Remove dep-info file generated by rustc. It is not tracked in
+                            // file_types. It does not have a prefix.
+                            (path_dash, ".d"),
+                            // Remove split-debuginfo files generated by rustc.
+                            (path_dot, ".o"),
+                            (path_dot, ".dwo"),
+                            (path_dot, ".dwp"),
+                        ];
+                        clean_ctx.rm_rf_prefix_list(&glob_pattern, &paths)?;
+                    }
+                }
+            }
+
+            // Clean incremental compilation files.
+            for target in pkg.targets() {
+                if target.is_custom_build() {
+                    continue;
+                }
+                let crate_name = target.crate_name();
+                for (_, layout) in &layouts {
+                    let incremental_dir = escape_glob_path(layout.build_dir().incremental())?;
+                    let pattern = Path::new(&incremental_dir).join(format!("{crate_name}-*"));
+                    clean_ctx.rm_rf_glob(&pattern)?;
+                }
```

Comment:
Why is this in a new loop body rather than the existing one like for the old layout?